device/coordinate_manager.h

```diff
@@ -0,0 +1,76 @@
+/*
+ * SPDX-FileCopyrightText: (c) 2023 Tenstorrent Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#pragma once
+
+#include <memory>
+#include <optional>
+#include <tuple>
+#include <vector>
+#include <map>
+#include <set>
+
+#include "device/tt_xy_pair.h"
+#include "device/tt_arch_types.h"
+#include <iostream>
+
+class CoordinateManager {
+
+public:
+    CoordinateManager(const tt_xy_pair& worker_grid_size, const std::vector<tt_xy_pair>& workers, std::size_t harvesting_mask)
+        : worker_grid_size(worker_grid_size), workers(workers), harvesting_mask(harvesting_mask) {}
+
+    virtual void perform_harvesting(std::size_t harvesting_mask);
+
+    virtual tt_physical_coords logical_to_physical_coords(tt_logical_coords logical_coords);
+    virtual tt_translated_coords logical_to_translated_coords(tt_logical_coords logical_coords);
```

Comment:
Consider naming this to_translated_coords(tt_logical_coords logical_coords); , e.g. without the "from" coordinates. Same for all functions. The to_translated_coords would then have 3 overrides for example.

Consider, and if you think this is better then feel free to leave it as it is.

Reply:
I like that suggestion. It would reduce the callers conditional checks based on the src coordinate type

Reply:
done